django/db/utils.py

```diff
@@ -197,6 +206,24 @@ def create_connection(self, alias):
         return backend.DatabaseWrapper(db, alias)
 
 
+class AsyncConnectionHandler(BaseAsyncConnectionHandler):
+    settings_name = "DATABASES"
+
+    def configure_settings(self, databases):
+        return configure_db_settings(super().configure_settings(databases))
+
+    def create_connection(self, alias):
+        db = self.settings[alias]
+        backend = load_backend(db["ENGINE"])
+
+        if not hasattr(backend, 'AsyncDatabaseWrapper'):
```

Comment:
If a database engine provides an AsyncDatabaseWrapper, it supports asynchronous connections. Otherwise, an error will be raised. This allows us to detect async support within the ORM.

You can check it like this:

```python
try:
    async_connections['default']
except ConnectionDoesNotExist:
    # Fallback to synchronous connection
```

I want to add a check for the presence of an event loop to avoid creating an async connection handler when no asynchronous environment is running. What do u think?